src/server/src/projects/entities/create-project.dto.ts

```diff
@@ -11,4 +11,6 @@ export class CreateProjectDto implements CreateProjectData {
   readonly numberOfDistricts: number;
   @IsNotEmpty({ message: "Need to supply a region configuration" })
   readonly regionConfig: RegionConfigIdDto;
+  @Allow()
```

Comment:
The validators on `update-project.dto.ts` should be able to be used here (and due to `@IsOptional` will be ignored if the `districtsDefinition` isn't provided)
```suggestion
  @IsArray()
  @ArrayNotEmpty()
  @IsOptional()
```

Reply:
Nice, I didn't know about `@IsOptional`. Makes sense.